Source/Core/SwipeActions.swift

```diff
@@ -0,0 +1,168 @@
+//
+//  Swipe.swift
+//  Eureka
+//
+//  Created by Marco Betschart on 14.06.17.
+//  Copyright © 2017 Xmartlabs. All rights reserved.
+//
+
+import Foundation
+
+public typealias SwipeActionHandler = (ContextualAction, ContextualActionSource, ((Bool) -> Void)?) -> Void
+
+public class SwipeAction{
+	public let contextualAction: ContextualAction
+	
+	@available(iOS 11, *)
+	public var backgroundColor: UIColor?{
```

Comment:
I would prefer to make `UIContextualAction` and `UITableViewRowAction ` conform to  a protocol like this:

```swift
public protocol ContextualAction{
     var backgroundColor: UIColor?
     var image: UIImage?
     var title: String?
}
```

Reply:
Of course 👍🏼 makes a whole lot more sense.

Reply:
Had to add the following in order to make `UITableViewRowAction` conform to the newly created `ContextualAction` protocol:

```
extension UITableViewRowAction: ContextualAction {
	public var image: UIImage? {
		get { return nil }
		set { return }
	}
}
```